doc/source/whatsnew/v0.25.0.rst

```diff
@@ -387,8 +387,8 @@ MultiIndex
 ^^^^^^^^^^
 
 - Bug in which incorrect exception raised by :class:`Timedelta` when testing the membership of :class:`MultiIndex` (:issue:`24570`)
--
--
+- Bug in having code value < -1
```

Comment:
can you make something like

``Bug in :class:`Multindex` construction from ``levels`` and ``codes`` that would incorrectly allows ``NaN`` levels ``